rclpy/src/rclpy/_rclpy_qos_event.c

```diff
@@ -30,14 +33,36 @@ static
 bool
 _check_rcl_return(rcl_ret_t ret, const char * error_msg)
 {
-  if (RCL_RET_OK != ret) {
-    PyErr_Format(
-      PyExc_RuntimeError,
-      "%s: %s", error_msg, rcl_get_error_string().str);
-    rcl_reset_error();
-    return false;
+  if (RCL_RET_OK == ret) {
+    return true;
+  }
+
+  PyObject * exception = PyExc_RuntimeError;
```

Comment:
This should be an `RCLError`, instead of `RuntimeError`.

Reply:
Done.